lib/ansible/modules/remote_management/dellemc/idrac/dellemc_idrac_firmware.py

```diff
@@ -0,0 +1,209 @@
+    from omsdk.sdkfile import FileOnShare
+    HAS_OMSDK = True
+except ImportError:
+    HAS_OMSDK = False
+
+
+def _validate_catalog_file(catalog_file_name):
+    normilized_file_name = catalog_file_name.lower()
+    if not normilized_file_name:
+        raise ValueError('catalog_file_name should be a non-empty string.')
+    elif not normilized_file_name.endswith("xml"):
+        raise ValueError('catalog_file_name should be an XML file.')
+
+
+def update_firmware(idrac, module):
+    """Update firmware from a network share and return the job details."""
+    msg = {}
+    msg['changed'] = False
+    msg['update_status'] = {}
+
+    try:
+        _validate_catalog_file(module.params['catalog_file_name'])
+    except ValueError as e:
+        module.fail_json(msg=str(e))
+
+    try:
+        upd_share = FileOnShare(remote=module.params['share_name'] + "/" + module.params['catalog_file_name'],
+                                mount_point=module.params['share_mnt'],
+                                isFolder=False,
+                                creds=UserCredentials(
+                                    module.params['share_user'],
+                                    module.params['share_pwd'])
+                                )
+
+        idrac.use_redfish = True
+        if '12' in idrac.ServerGeneration or '13' in idrac.ServerGeneration:
+            idrac.use_redfish = False
+
+        apply_update = True
+        msg['update_status'] = idrac.update_mgr.update_from_repo(upd_share,
+                                                                 apply_update,
+                                                                 module.params['reboot'],
+                                                                 module.params['job_wait'])
+    except Exception as e:
+        module.fail_json(msg=str(e))
+
+    if "Status" in msg['update_status']:
+        if msg['update_status']['Status'] == "Success":
+            if module.params['job_wait'] is True:
+                msg['changed'] = True
+        else:
+            module.fail_json(msg=msg['update_status'])
+    return msg
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec={
+            "idrac_ip": {"required": True, "type": str},
+            "idrac_user": {"required": True, "type": str},
+            "idrac_pwd": {"required": True, "type": str, "no_log": True},
+            "idrac_port": {"required": False, "default": 443, "type": int},
+
+            "share_name": {"required": True, "type": str},
+            "share_user": {"required": False, "type": str},
+            "share_pwd": {"required": False, "type": str, "no_log": True},
+            "share_mnt": {"required": True, "type": str},
+
+            "catalog_file_name": {"required": False, "type": str, "default": "Catalog.xml"},
+            "reboot": {"required": False, "type": bool, "default": False},
+            "job_wait": {"required": False, "type": bool, "default": True}
```

Comment:
Could you please add a trailing comma here?
```suggestion
            "job_wait": {"required": False, "type": bool, "default": True},
```

Reply:
sure .. done